src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -90,7 +111,8 @@ void M5(S<T> s5)
         M2(ref s5.F1);
     }
 }";
-            var comp = CreateCompilation(sourceA, parseOptions: TestOptions.Regular10);
+            var mscorlibWithRefFields = MscorlibRefWithoutSharingCachedSymbols;
+            var comp = CreateByRefFieldsCompilation(sourceA, references: new[] { mscorlibWithRefFields }, parseOptions: TestOptions.Regular10);
```

Comment:
>CreateByRefFieldsCompilation(sourceA, references: new[] { mscorlibWithRefFields }, parseOptions: TestOptions.Regular10) [](http://example.com/codeflow?start=23&length=119)

Could we hide most of the work in `CreateCompilation()`?

For instance, add a parameter to `CreateCompilation(..., RuntimeFeatures runtimeFeatures, ...)` that creates an cloned version of the appropriate corlib with the necessary features enabled, and that cloned version is created at most once and shared with all callers that need the same set of features.